python/desc/imsim/ImageSimulator.py

```diff
@@ -284,16 +288,44 @@ def __call__(self, gs_objects):
             os.makedirs(outdir)
         prefix = image_simulator.config['persistence']['eimage_prefix']
         obsHistID = str(image_simulator.obs_md.OpsimMetaData['obshistID'])
-        gs_interpreter.writeImages(nameRoot=os.path.join(outdir, prefix)
-                                   + obsHistID)
+        nameRoot = os.path.join(outdir, prefix) + obsHistID
+        outfiles = gs_interpreter.writeImages(nameRoot=nameRoot)
+        if image_simulator.config['persistence']['eimage_compress']:
+            compress_files(outfiles)
 
         # Write out the centroid files if they were made.
         gs_interpreter.write_centroid_files()
 
-#        # The image for the sensor-visit has been drawn, so delete the
-#        # checkpoint file.
-#        os.remove(gs_interpreter.checkpoint_file)
+        # The image for the sensor-visit has been drawn, so delete any
+        # existing checkpoint file if the config says to do so.
+        if (gs_interpreter.checkpoint_file is not None
+            and os.path.isfile(gs_interpreter.checkpoint_file)
+            and image_simulator.config['checkpointing']['cleanup']):
+            os.remove(gs_interpreter.checkpoint_file)
```

Comment:
This is really adding the ability to *not* delete a checkpoint correct?  I think it was already deleting the checkpoint after the image was written before.  Is there any other feature being added here (just a bit confused since the main PR says it adds the ability per Adrian's request).

Reply:
The code being replaced has the call to remove the checkpoint file commented out (and in fact would crash if the checkpointing option were not enabled at runtime).  So this change does enable the checkpoint file to be deleted, *if* it exists and if the config file says it should be.